tgui/packages/tgui/interfaces/HealthScan.js

```diff
@@ -0,0 +1,457 @@
+import { useBackend } from '../backend';
+import { Section, ProgressBar, Box, LabeledList, NoticeBox, Stack, Icon, Divider } from '../components';
+import { Window } from '../layouts';
+
+export const HealthScan = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    patient,
+    dead,
+    health,
+    total_brute,
+    total_burn,
+    toxin,
+    oxy,
+    clone,
+
+    has_chemicals,
+    has_unknown_chemicals,
+    chemicals_lists,
+
+    limb_data_lists,
+    limbs_damaged,
+
+    damaged_organs,
+    ssd,
+
+    diseases,
+
+    blood_type,
+    blood_amount,
+    has_blood,
+    body_temperature,
+    pulse,
+    internal_bleeding,
+    implants,
+    core_fracture,
+    lung_ruptured,
+    hugged,
+    detail_level,
+    permadead,
+    advice,
+    species,
+  } = data;
+  const chemicals = Object.values(chemicals_lists);
+  const limb_data = Object.values(limb_data_lists);
+  const healthanalyser = detail_level < 1;
+  const bodyscanner = detail_level >= 1;
+  const ghostscan = detail_level >= 2;
+
+  const Synthetic = species === "Synthetic";
+
+  const theme = Synthetic ? "hackerman" : (bodyscanner ? "ntos" : "default");
+
+  return (
```

Comment:
Whilst generally fine I would recommend a few things:

The component is several hundred lines of code. I would generally recommend separating out parts of the code to just make it easier to see what is going on. For example, you can render a single component to handle the mapping of `limb_data` so `map(x => <LimbRender limb={x}/>`.

Also, it's generally a good idea to style components separately rather than styling components explicitly. For example:

```
<Box inline
  color={organ.status === "Bruised" ? "orange" : "red"}
  bold={1}>
  {organ.status + " [" + organ.damage + " damage]"}
</Box>
```

can become
```
<Box inline className="HealthAnalyser_organ_damage">
  {organ.status + " [" + organ.damage + " damage]"}
</Box>
```

This allows for a good level of customisation between different themes for colouring.

Reply:
first part done